packages/react/src/profile/useCreateProfile.ts

```diff
@@ -1,23 +1,46 @@
-import { DuplicatedHandleError } from '@lens-protocol/domain/use-cases/profile';
+import { TransactionError, TransactionKind } from '@lens-protocol/domain/entities';
+import {
+  CreateProfileRequest,
+  DuplicatedHandleError,
+} from '@lens-protocol/domain/use-cases/profile';
 import { useState } from 'react';
 
+import {
+  TransactionState,
+  useWaitUntilTransactionSettled,
+} from '../transactions/adapters/TransactionQueuePresenter';
 import { useCreateProfileController } from './adapters/useCreateProfileController';
 
 export function useCreateProfile() {
-  const [error, setError] = useState<DuplicatedHandleError | null>(null);
-  const [isPending, setIsPending] = useState(false);
+  const [error, setError] = useState<DuplicatedHandleError | TransactionError | null>(null);
+  const [isPending, setIsPending] = useState<boolean>(false);
+
   const createProfile = useCreateProfileController();
 
+  const waitUntilTransactionIsSettled = useWaitUntilTransactionSettled();
+
   return {
     create: async (handle: string) => {
-      setIsPending(true);
-
       try {
+        setIsPending(true);
+        setError(null);
+
         const result = await createProfile(handle);
 
         if (result.isFailure()) {
           setError(result.error);
         }
+
+        await waitUntilTransactionIsSettled(
```

Comment:
@krzysu this allows us to wait until the tx has reached settled status so that we can automatically show the created profile in the UI